### PR TITLE
phpmd fixes

### PR DESCRIPTION
**Pull request by simon-letch**
PR for a few of the phpmd fixes. Just to make sure i'm along the right lines. I've mainly used phpstorm's refactor tool to extract methods.

### CODE REVIEW

File: Model/Sync/Contact/Update.php [hunk 1]
```diff
@@ -53,71 +53,109 @@ public function sync($collection)
...
+            $this->_handleSingleItemAfterSync($item, $result);
+        }
+    }
+
+    /**
+     * @param $importData
+     * @param $websiteId
+     * @return mixed
+     */
+    public function syncItemContactEmailUpdateMode($importData, $websiteId)
```
Comment (cdiacon) [inline Model/Sync/Contact/Update.php]: change to private

File: Model/Sync/Contact/Update.php [hunk 2]
```diff
@@ -53,71 +53,109 @@ public function sync($collection)
...
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * @param $importData
+     * @return mixed
+     */
+    public function syncItemSubscriberResubscribedMode($importData)
```
Comment (cdiacon) [inline Model/Sync/Contact/Update.php]: change to private

File: Model/Sync/Contact/Update.php [hunk 3]
```diff
@@ -53,71 +53,109 @@ public function sync($collection)
...
+        }
+        return $result;
+    }
+
+    /**
+     * @param $importData
+     * @param $websiteId
+     * @return mixed
+     */
+    public function syncItemSubscriberUpdateMode($importData, $websiteId)
```
Comment (cdiacon) [inline Model/Sync/Contact/Update.php]: change to private

File: Model/Trial/TrialSetup.php [hunk 4]
```diff
@@ -239,4 +214,41 @@ public function saveApiEndPoint($value)
             0
         );
     }
+
+    /**
+     * @param $client
+     * @param $addressBooks
+     * @return bool
+     */
+    public function validateAccountAndCreateAddressbooks($client, $addressBooks)
```
Comment (cdiacon) [inline Model/Trial/TrialSetup.php]: change to private

File: Observer/Catalog/ResetCatalogContactImport.php [hunk 5]
```diff
@@ -51,53 +51,70 @@ public function __construct(
...
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return array
+     */
+    public function resetConnectorCatalogFactoryIfRequired(\Magento\Framework\Event\Observer $observer)
```
Comment (cdiacon) [inline Observer/Catalog/ResetCatalogContactImport.php]: change to private

File: Observer/Catalog/ResetCatalogContactImport.php [hunk 6]
```diff
@@ -51,53 +51,70 @@ public function __construct(
...
-                                ->resetAllContacts();
-                        }
-                        $this->registry->register(
-                            'core_config_data_save_after_done_status',
-                            true
-                        );
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     */
+    public function resetAllContactsInConnectorCatalogFactoryIfRequired(\Magento\Framework\Event\Observer $observer)
```
Comment (cdiacon) [inline Observer/Catalog/ResetCatalogContactImport.php]: change to private

File: Observer/Catalog/ResetCatalogContactImport.php [hunk 7]
```diff
@@ -51,53 +51,70 @@ public function __construct(
...
+        } catch (\Exception $e) {
+            $this->helper->debug((string)$e, []);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return array
```
Comment (cdiacon) [inline Observer/Catalog/ResetCatalogContactImport.php]: remove return array

File: Test/Integration/Automation/FirstOrderTest.php [hunk 8]
```diff
@@ -32,7 +32,7 @@ public function setUp()
      */
     public function test_first_customer_automation()
     {
-        $customer = $this->objectManager->create(\Magento\Customer\Model\Customer::class);
+        $this->objectManager->create(\Magento\Customer\Model\Customer::class);
```
Comment (cdiacon) [inline Test/Integration/Automation/FirstOrderTest.php]: remove  use Magento\TestFramework\ObjectManager; L5

File: Model/Sync/Contact/Update.php (hunk 1 above)
Comment (simon-letch) [inline Model/Sync/Contact/Update.php]: Fixed

File: Model/Sync/Contact/Update.php (hunk 2 above)
Comment (simon-letch) [inline Model/Sync/Contact/Update.php]: Fixed

File: Model/Sync/Contact/Update.php (hunk 3 above)
Comment (simon-letch) [inline Model/Sync/Contact/Update.php]: Fixed

File: Model/Trial/TrialSetup.php (hunk 4 above)
Comment (simon-letch) [inline Model/Trial/TrialSetup.php]: Fixed

File: Observer/Catalog/ResetCatalogContactImport.php (hunk 7 above)
Comment (simon-letch) [inline Observer/Catalog/ResetCatalogContactImport.php]: Fixed

File: Observer/Catalog/ResetCatalogContactImport.php (hunk 5 above)
Comment (simon-letch) [inline Observer/Catalog/ResetCatalogContactImport.php]: Fixed

File: Observer/Catalog/ResetCatalogContactImport.php (hunk 6 above)
Comment (simon-letch) [inline Observer/Catalog/ResetCatalogContactImport.php]: Fixed

File: Test/Integration/Automation/FirstOrderTest.php (hunk 8 above)
Comment (simon-letch) [inline Test/Integration/Automation/FirstOrderTest.php]: Fixed

**Comment by simon-letch** [comment]
@cdiacon i've fixed your points and updated the PR

**Pull request merged**